File: minwook/8주차/크기가_작은_부분_문자열.java
class Solution {
    public int solution(String t, String p) {
        int answer = 0;
        String[] tArr = t.split("");
        int n = 1;

        for(int i=0; i<=tArr.length-p.length(); i++){
            while(tArr[i].length() < p.length()){
                tArr[i] += tArr[i+n];
                n++;
            }
            n=1;
            if(Long.parseLong(tArr[i]) <= Long.parseLong(p))
                answer++; 
        }
        return answer;
    }
}

// t에 있는 문자열 정수를 p의 문자열 정수 자릿수만큼 잘라서 값을 비교하여 도출

File: yongkeun/8week/크기가작은부분문자열.js
function solution(t, p) {
  answer = [];
  count = 0;
  t = t.split(""); //배열로 바꾸고
  for (let i = 0; i <= t.length - p.length; i++) {
    //p의 길이가 3이고 t가 123456이면 123 234 345 456을 배열에 푸시
    answer.push(t.slice(i, i + p.length).join(""));
  }
  answer.forEach((value) => {
    if (value <= p)
      //배열의 값이 p보다 작거나 같으면 count 더함
      count++;
  });
  return count;
}
